common/lib/xmodule/xmodule/modulestore/mongo/base.py

```diff
@@ -980,6 +981,40 @@ def _should_apply_cached_metadata(self, item, depth):
         return apply_cached_metadata
 
     @autoretry_read()
+    def get_course_summaries(self, **kwargs):
+        """
+        Returns a list of `CourseSummary`. This accepts an optional parameter of 'org' which
+        will apply an efficient filter to only get courses with the specified ORG
+        """
+        def extract_course_summary(course):
+            """
+            Extract course information from the course block for mongo.
+            """
+            return {
+                field: course['metadata'][field]
+                for field in CourseSummary.course_info_fields
+                if field in course['metadata']
+            }
```

Comment:
I done it this way, because it was much more readable.While iterating our courses we are using a helper method `extract_course_summary` which extracts the relevant course summary of a course block. You are right we are only using it once.